### ISSUE TITLE
delete command should be listed in 'cf -h' 

### ISSUE THREAD

**Issue by gidsg** [cloudfoundry/cli#1311]
## Command  
`cf -h`

## What occurred

`delete` is not listed under `Application lifecycle`

## What you expected to occur

`delete` is listed under `Application lifecycle`

## CLI Version  

cf version 6.34.1+bbdf81482.2018-01-17

## CC API Endpoint Version  

N/A

## Any other relevant information  

It is an anomaly that other delete options are listed such as `delete-route` but not [delete](http://cli.cloudfoundry.org/en-US/cf/delete.html)

I am happy to work on a patch to add this if you agree it should be implemented.

**Comment by dkoper** [comment]
Hi @gidsg, @jaydunk 

We intentionally don't display all commands. We used to, and `cf help` would result in several pages of blobs of text scrolling over the screen and without grep it was impossible to find anything fast.

We addressed that by reducing cf help to only show the most commonly used app developer commands. The rest is available under `cf help -a`.

Our assumption was that the `delete` command was rarely used. It would definitely be less used than `push`, as people do not need to delete an old app to push an update. Also, the point of help, and the reduced help page, is to assist with quick discovery. I assumed that `delete` is so common with command names like `start` and `stop` around that no-one (until now?) has contacted us in despair not being able to find the command to delete an app. That is the core of your issue, isn't it?

There are definitely inconsistencies - the decision for each command to go in `cf help` was not based on empirical data, but I would remind people who want to solve these inconsistencies by adding such commands back to help, that that could take us back to the place where we were 18 months ago,

That said, I can't remember why we included `delete-route` in `cf help` - removing it may be a better alternative?
